apps/sharedtest/test/unit/sync/sync-engine/syncengine_test.js

```diff
@@ -288,10 +288,40 @@ ld be a Function`);
       });
     });
 
+    test('repeats as long as fetchToken is returned', function(done) {
+      var credentials = {
+        URL: 'http://localhost:8000/v1/',
+        assertion: 'test-assertion-mock',
+        kB: '85c4f8c1d8e3e2186824c127af786891dd03c6e05b1b45f28f7181211bf2affb',
+        adapters: {
+          bookmarks: {
+            update: function() { },
+            handleConflict: function() { },
+            reset: function() { }
+          }
+        }
+      };
+      var se = new SyncEngine(credentials);
+      for (var i = 0; i < 8; i++) {
+        FxSyncWebCrypto.responses.push({ foo: i });
+      }
+      se.syncNow({ 'bookmarks': {
+        limit: 3
```

Comment:
Note that `limit` will be renamed to `fetchLimit` in https://github.com/mozilla-b2g/gaia/pull/33745/files#diff-795a9307cc60d72cc4ceaa37fddaf7b5R279